src/suites/cts/validation/setScissorRect.spec.ts

```diff
@@ -0,0 +1,87 @@
+export const description = `
+setScissorRect validation tests.
+`;
+
+import { TestGroup } from '../../../framework/index.js';
+
+import { ValidationTest } from './validation_test.js';
+
+// TODO: Move this fixture class to a common file.
+export class F extends ValidationTest {
+  textureWidth: number = 16;
+  textureHeight: number = 16;
+
+  beginRenderPass(commandEncoder: GPUCommandEncoder): GPURenderPassEncoder {
+    const attachmentTexture = this.device.createTexture({
+      format: 'rgba8unorm',
+      size: { width: this.textureWidth, height: this.textureHeight, depth: 1 },
+      usage: GPUTextureUsage.OUTPUT_ATTACHMENT,
+    });
+
+    return commandEncoder.beginRenderPass({
+      colorAttachments: [
+        {
+          attachment: attachmentTexture.createView(),
+          loadValue: { r: 1.0, g: 0.0, b: 0.0, a: 1.0 },
+        },
+      ],
+    });
+  }
+}
+
+export const g = new TestGroup(F);
+
+g.test('basic use of setScissorRect', t => {
+  const commandEncoder = t.device.createCommandEncoder();
+  const renderPass = t.beginRenderPass(commandEncoder);
+  renderPass.setScissorRect(0, 0, 1, 1);
+  renderPass.endPass();
+  commandEncoder.finish();
+});
```

Comment:
While we're at it, let's also add more cases, like the ones you have for viewport below.

Reply:
> Let's parameterize these tests, e.g. `{rect: [0, 0, 1, 1], success: true}`
> 
> Note I added a second argument to expectValidationError, shouldError, to help with this kind of parameterization.

Done.

Reply:
> While we're at it, let's also add more cases, like the ones you have for viewport below.

The ones I have for viewport make sense because values are `float`.
In this case, they are `unsigned long`: an unsigned integer type that has values in the range [0, 4294967295].

Reply:
OK, we'll do the IDL type check tests somewhere else later.